From 6e9e103d18bd8459c7bbed14fcf7161a0fb9291c Mon Sep 17 00:00:00 2001
From: lyz <lyz@rock-chips.com>
Date: Tue, 9 Dec 2014 22:23:05 +0800
Subject: [PATCH] rk312x: usb: Skip DWC HOST for rk3126/3128 and usb phy tuning

1.Use EHCI/OHCI for all rk312x series
2.Set Host disconnect trigger point to 600mv, default is 650mv
---
 drivers/usb/dwc_otg_310/usbdev_rk3126.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/dwc_otg_310/usbdev_rk3126.c b/drivers/usb/dwc_otg_310/usbdev_rk3126.c
index a6f174cc322d..5b7ba89aea94 100755
--- a/drivers/usb/dwc_otg_310/usbdev_rk3126.c
+++ b/drivers/usb/dwc_otg_310/usbdev_rk3126.c
@@ -8,7 +8,7 @@ static void usb20otg_hw_init(void)
 {
 	/* Turn off differential receiver in suspend mode */
 	writel(UOC_HIWORD_UPDATE(0, 1, 2),
-		   RK_GRF_VIRT + RK312X_GRF_USBPHY1_CON6);
+		   RK_GRF_VIRT + RK312X_GRF_USBPHY0_CON6);
 	/* other haredware init,include:
 	 * DRV_VBUS GPIO init */
 	if (gpio_is_valid(control_usb->otg_gpios->gpio)) {
@@ -216,12 +216,12 @@ struct dwc_otg_platform_data usb20otg_pdata_rk3126 = {
 #ifdef CONFIG_USB20_HOST
 static void usb20host_hw_init(void)
 {
-	/* Switch to DWC HOST */
-	writel(UOC_HIWORD_UPDATE(1, 1, 3),
-		   RK_GRF_VIRT + RK312X_GRF_SOC_CON2);
 	/* Turn off differential receiver in suspend mode */
 	writel(UOC_HIWORD_UPDATE(0, 1, 2),
-		   RK_GRF_VIRT + RK312X_GRF_USBPHY0_CON6);
+		   RK_GRF_VIRT + RK312X_GRF_USBPHY1_CON6);
+	/* Set disconnect detection trigger point to 600mv */
+	writel(UOC_HIWORD_UPDATE(1, 0xf, 11),
+		   RK_GRF_VIRT + RK312X_GRF_USBPHY1_CON7);
 	/* other haredware init,include:
 	 * DRV_VBUS GPIO init */
 	if (gpio_is_valid(control_usb->host_gpios->gpio)) {
-- 
2.35.3

